src/hooks.js

```diff
@@ -0,0 +1,41 @@
+import { useState, useEffect } from 'preact/hooks'
+
+const fetchBody = async (url, fetchOpts) => {
+  const state = {}
+  try {
+    state.response = await fetch(url, fetchOpts)
+    state.body = await state.response.text()
+    state.error = state.response.ok
+      ? undefined
+      : Error(state.response.statusText)
+  } catch (error) {
+    state.error = error
+  }
+  return state
+}
+
+export const useFetchJSON = (...args) => {
+  const state = useFetch(...args)
+  if (!state.body) return state
+  try {
+    state.data = JSON.parse(state.body)
+  } catch (error) {
+    state.error = error
+  }
+  return state
+}
+
+export const useFetch = (url, fetchOpts, inputs = [url]) => {
```

Comment:
this is where we make the link with the react hooks api.

We want to get the data when the hook is called, for that we need a state and an effect